File: java/src/test/java/problems201_300/MoveZeroesTest.java
package problems201_300;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MoveZeroesTest extends SampledTest {
    @Test
    public void test() {
        for (MoveZeroes solution : MoveZeroes.values()) {
            for (MoveZeroesSample sample : getSamples(MoveZeroesSample[].class)) {
                int[] nums = sample.input.nums;
                solution.moveZeroes(nums);
                assertWithMessage(sample.getMessage())
                    .that(nums)
                    .asList()
                    .containsExactlyElementsIn(Arrays.stream(sample.output).boxed().toList())
                    .inOrder();
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/MoveZeroesTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class MoveZeroesTest : SampledTest() {
    @Test
    fun test() =
        MoveZeroes.entries.forEach { solution ->
            getSamples<Array<MoveZeroesSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(input.nums.also { solution.moveZeroes(it) })
                    .asList()
                    .containsExactlyElementsIn(output.toList())
                    .inOrder()
            }
        }
}
